docs/documentation/helm.md

```diff
@@ -0,0 +1,277 @@
+- The Kubernetes ingress host
+- The Openshift S2i builder image
+
+For example, if you set 3 replicas for your deployment:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+
+# Set replicas to 3
+dekorate.kubernetes.replicas=3
+```
+
+Dekorate will generate the next Helm values file at `target/classes/META-INF/dekorate/helm/<chart name>/values.yaml`:
+
+```yaml
+---
+myModule:
+  replicas: 3
+```
+
+Note that `myModule` is the name of your project.
+
+And the Deployment file at `target/classes/META-INF/dekorate/helm/<chart name>/templates/deployment.yaml` will have a reference to this value:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myModule
+spec:
+  replicas: {{ .Values.myModule.replicas }}
+```
+
+This is done transparently to users.
+
+##### Mapping user properties using JSONPath expressions
+
+What about if you want to map another property like the name of all the resources (the metadata name)?
+Dekorate allows users to define [JSONPath](https://tools.ietf.org/id/draft-goessner-dispatch-jsonpath-00.html) expressions to map properties into the Helm values file.
+
+For example, having the following YAML resource:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: helm-on-kubernetes-example
+...
+```
+
+After getting familiar with JSONPath, the expression to map the metadata name value is `$..metadata.name`, so you need to add it to your configuration:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+
+# Map all the metadata name resources
+dekorate.helm.values[0].property=myModule.name
+dekorate.helm.values[0].jsonPaths=$..metadata.name
+```
+
+| Note that JSONPath works with JSON, not YAML. So, for testing your expressions, you first need to convert YAML files to JSON files using [this online tool](https://jsonformatter.org/yaml-to-json) and then use some JSONPath online evaluator like [https://jsonpath.com/](https://jsonpath.com/). Remember to wrap the json objects into lists.
+
+The resulting `values.yaml` file will look like as:
+
+```yaml
+myModule:
+  name: helm-on-kubernetes-example
+```
+
+Why the value is `helm-on-kubernetes-example`? This is because Dekorate will automatically inspect the generated resources to find the matching value of the JSONPath expression. 
+
+However, users can provide other values using the `value` property:
+
+```
+dekorate.helm.name=myChart
+dekorate.helm.description=Description of my Chart
+
+# Map all the metadata name resources
+dekorate.helm.values[0].property=myModule.name
+dekorate.helm.values[0].jsonPaths=$..metadata.name
+dekorate.helm.values[0].value=this-is-another-name
+```
+
+And the `values.yaml` file will now contain:
+
+```yaml
+myModule:
+  name: this-is-another-name
+```
+
+##### Mapping multiple properties at once
+
+What about if the properties to map are at different locations, for example:
```

Comment:
Rephrase to `are located in different places` AND better detail what is the purpose of this example. What do you want to find and replace ...

Reply:
Changed